lib/mac/yosemite.js

```diff
@@ -0,0 +1,402 @@
+
+var uuidToAddress = require('./uuid-to-address');
+
+var osRelease = parseFloat(os.release());
+
+var BlenoBindings = function() {
+  this._xpcConnection = new XpcConnection('com.apple.blued');
+  this._deviceUUID = null;
+
+  this._xpcConnection.on('error', function(message) {
+    this.emit('xpcError', message);
+  }.bind(this));
+
+  this._xpcConnection.on('event', function(event) {
+    this.emit('xpcEvent', event);
+  }.bind(this));
+};
+
+util.inherits(BlenoBindings, events.EventEmitter);
+
+BlenoBindings.prototype.sendXpcMessage = function(message) {
+  this._xpcConnection.sendMessage(message);
+};
+
+BlenoBindings.prototype.disconnect = function() {
+  throw new Error('disconnect is not supported on OS X!');
+};
+
+var blenoBindings = new BlenoBindings();
+
+blenoBindings.on('xpcEvent', function(event) {
+  var kCBMsgId = event.kCBMsgId;
+  var kCBMsgArgs = event.kCBMsgArgs;
+
+  debug('xpcEvent: ' + JSON.stringify(event, undefined, 2));
+
+  this.emit('kCBMsgId' + kCBMsgId, kCBMsgArgs);
+});
+
+blenoBindings.on('xpcError', function(message) {
+  console.error('xpcError: ' + message);
+});
+
+blenoBindings.sendCBMsg = function(id, args) {
+  debug('sendCBMsg: ' + id + ', ' + JSON.stringify(args, undefined, 2));
+  this.sendXpcMessage({
+    kCBMsgId: id,
+    kCBMsgArgs: args
+  });
+};
+
+blenoBindings.init = function() {
+  this._xpcConnection.setup();
+
+  child_process.exec('system_profiler SPBluetoothDataType', {}, function(error, stdout, stderr) {
+    this.emit('platform', os.platform());
+
+    if (!error) {
+      var found = stdout.match(/\s+Address: (.*)/);
+      if (found) {
+        var address = found[1].toLowerCase().replace(/-/g, ':');
+
+        this.emit('addressChange', address);
+      }
+    }
+
+    if (osRelease < 13) {
+      debug('bleno warning: OS X < 10.9 detected');
+
+      console.warn('bleno requires OS X 10.9 or higher!');
+
+      this.emit('stateChange', 'unsupported');
+    } else {
+      this.sendCBMsg(1, {
+        kCBMsgArgName: 'node-' + (new Date()).getTime(),
+        kCBMsgArgOptions: {
+            kCBInitOptionShowPowerAlert: 1
+        },
+        kCBMsgArgType: 1
+      });
+    }
+  }.bind(this));
+};
+
+blenoBindings.on('kCBMsgId6', function(args) {
+  var state = ['unknown', 'resetting', 'unsupported', 'unauthorized', 'poweredOff', 'poweredOn'][args.kCBMsgArgState];
+  debug('state change ' + state);
+  this.emit('stateChange', state);
+});
+
+blenoBindings.startAdvertising = function(name, serviceUuids) {
+  var advertisement = {
+    kCBAdvDataLocalName: name,
+    kCBAdvDataServiceUUIDs: []
+  };
+
+  if (serviceUuids && serviceUuids.length) {
+    for(var i = 0; i < serviceUuids.length; i++) {
+      advertisement.kCBAdvDataServiceUUIDs[i] = new Buffer(serviceUuids[i], 'hex');
+    }
+  }
+
+  this.sendCBMsg(8, advertisement);
+};
+
+blenoBindings.startAdvertisingIBeacon = function(data) {
+  var args = {};
+
+  if (osRelease >= 14) {
```

Comment:
@sandeepmistry This one too, no?

Reply:
This is file is for Yosemite -> Sierra, we should leave everything in this file as is.

Reply:
ah hell, not this one.